scaleway/resource_domain_record.go

```diff
@@ -261,11 +263,45 @@ func resourceScalewayDomainRecordCreate(ctx context.Context, d *schema.ResourceD
 		return diag.FromErr(err)
 	}
 
-	_, err = waitForDNSZone(ctx, domainAPI, dnsZone, d.Timeout(schema.TimeoutCreate))
+	dnsZoneRes, err := waitForDNSZone(ctx, domainAPI, dnsZone, d.Timeout(schema.TimeoutCreate))
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
+	tflog.Debug(ctx, fmt.Sprintf("DNS ZONE domain: %s subdomain: %s, status: %s",
+		dnsZoneRes.Domain,
+		dnsZoneRes.Subdomain,
+		dnsZoneRes.Status))
+
+	dnsZoneData, err := domainAPI.ListDNSZoneRecords(&domain.ListDNSZoneRecordsRequest{
+		DNSZone: dnsZone,
+		Name:    d.Get("name").(string),
+		Type:    recordType,
+	}, scw.WithAllPages(), scw.WithContext(ctx))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var currentRecord *domain.Record
```

Comment:
Could this be extracted in a function?

Reply:
just added on helpers`getRecordFromData`